### PR TITLE
Issue #59: Implement script/contract for random number generator

### LINKED ISSUE #59: Implement script/contract for random number generator
# Implement Random Number Generator

## Objective
Create a script/contract that generates random numbers using different techniques as you decide.

## Techniques to consider for this implementation
- Generator using block hash
- Generator using keccak256
- Combination of multiple entropy sources
- Pragma Verifiable Random Function (VRF): https://docs.pragma.build/v1/Resources/randomness/randomness

## Read for more understanding
- https://starknet-by-example.voyager.online/applications/random_number_generator

## 📜 Additional Notes
⚠ **Do not apply until the ODHack begins.**
⚠ **Read our guidelines before applying.**

### PR DESCRIPTION
**Pull request by jorgezerpa**
Closes #59 

### Create a contract RandomNumberGeneratorContract and it's tests. 

The contract has 4 examples of how to generate random numbers:
- get_random_number_with_block_data
- get_random_number_with_keccak256
- get_random_number_with_user_entropy
- get_random_number_with_vrf & receive_random_words 

The first 3 methods has an input that applies a bit of entropy to "ensure" randomness but are not recommended since such randomness is not verifiable. 
The last method uses the Pragma VRF functionality to get the number. the get_random_number_with_vrf request a random number and the receive_random_words is the function used as callback by the VRF contract to return the random number. 

I'm attentive to any correction/question🚀🚀

### CODE REVIEW

**Review by coxmars — COMMENTED** [review]
@jorgezerpa add a .tools-versions file in the project "starknet/contracts/random_number_generator" to document your dependencies.

**Comment by coxmars** [comment]
@jorgezerpa please send an image of your tests results to validate all of them are passing and it is working as expected 🫡

**Comment by jorgezerpa** [comment]
@coxmars Hello! I do it super fast🏃🏃

**Comment by jorgezerpa** [comment]
.tools-versions file added.
Here's the image:  of tests: 
image

**Comment by coxmars** [comment]
> .tools-versions file added. Here's the image: of tests: image

Awesome job sir, I will review it 🫡

File: starknet/contracts/random_number_generator/ .tools-versions
Comment (danielcdz) [inline starknet/contracts/random_number_generator/ .tools-versions:1]: this file needs to be named .tool-versions to work properly, can you please rename it?

**Review by danielcdz — CHANGES_REQUESTED** [review]

**Comment by jorgezerpa** [comment]
@danielcdz Ready both changes sir!

File: starknet/contracts/random_number_generator/.tool-versions
Comment (danielcdz) [inline starknet/contracts/random_number_generator/.tool-versions:1]: this file needs an extra line at the end, then we can merge! 🚀

Comment (jorgezerpa) [reply starknet/contracts/random_number_generator/.tool-versions:1]: Done this here🥷

**Review by danielcdz — APPROVED** [review]

**Pull request merged**